tests/integration/test_join.py

```diff
@@ -402,6 +403,50 @@ def test_filter_columns_post_join(c):
     assert_eq(result_df, expected_df)
 
 
+def test_join_reorder(c):
```

Comment:
Could we add some tests around other potential join cases we would run into? In particular maybe some with:

- unsupported join types / conditions
- joins that would result in more than 2 dominant fact tables
-

Reply:
The logic for joins involving different combinations of numbers of fact tables versus numbers of dimension tables is a little shaky (especially depending on the number of fact tables and fact-fact joins), so I'd prefer to hold off there until I have a better solution for multiple fact tables. I have this listed in https://github.com/dask-contrib/dask-sql/issues/1069

I can work on examples for unsupported join types and conditions, though.